diff --git a/Modules/ThirdParty/GDCM/CMakeLists.txt b/Modules/ThirdParty/GDCM/CMakeLists.txt
--- a/Modules/ThirdParty/GDCM/CMakeLists.txt
+++ b/Modules/ThirdParty/GDCM/CMakeLists.txt
@@ -4,7 +4,7 @@ if(ITK_USE_SYSTEM_GDCM)
 if(ITK_USE_SYSTEM_GDCM)
   set(ITKGDCM_SYSTEM_INCLUDE_DIRS ${GDCM_INCLUDE_DIRS})
   set(ITKGDCM_SYSTEM_LIBRARY_DIRS ${GDCM_LIBRARY_DIRS})
-  set(ITKGDCM_LIBRARIES gdcmDICT gdcmMSFF)
+  set(ITKGDCM_LIBRARIES gdcmDICT gdcmMSFF gdcmDSED)
   set(ITKGDCM_NO_SRC 1)

   # When this module is loaded by an app, load GDCM too.
@@ -42,7 +42,7 @@ else()
       endif()
     endif()
   endif()
-  set(ITKGDCM_LIBRARIES  gdcmDICT gdcmMSFF )
+  set(ITKGDCM_LIBRARIES  gdcmDICT gdcmMSFF gdcmDSED)
 endif()

 itk_module_impl()
diff --git a/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt b/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt
--- a/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt
+++ b/Modules/ThirdParty/Expat/src/expat/CMakeLists.txt
@@ -38,5 +38,6 @@ INSTALL(TARGETS ITKEXPAT
 INSTALL(FILES
   ${ITK3P_EXPAT_BINARY_DIR}/expatDllConfig.h
   ${ITK3P_EXPAT_SOURCE_DIR}/expat.h
+  ${ITK3P_EXPAT_SOURCE_DIR}/itk_expat_mangle.h
   DESTINATION ${ITK3P_INSTALL_INCLUDE_DIR} # TODO: itk_expat.h #include "itkexpat/expat.h"
   COMPONENT Development)
diff --git a/Modules/IO/PhilipsREC/src/itkPhilipsPAR.cxx b/Modules/IO/PhilipsREC/src/itkPhilipsPAR.cxx
index 68b09ea..4cdb36b 100644
--- a/Modules/IO/PhilipsREC/src/itkPhilipsPAR.cxx
+++ b/Modules/IO/PhilipsREC/src/itkPhilipsPAR.cxx
@@ -1223,6 +1223,7 @@ void PhilipsPAR::ReadPAR(std::string parFile, struct par_parameter *pPar)
       pPar->vox[0] = tempInfo.spacingx;
       pPar->vox[1] = tempInfo.spacingy;
       pPar->vox[2] = tempInfo.slice_thick + tempInfo.slice_gap;
+	  pPar->cardiac_freq = tempInfo.cardiac_freq;
       inString.str( this->GetGeneralInfoString(parFile, 31) );
       inString >> fovAP >> fovFH >> fovRL;
       inString.clear();
@@ -1560,6 +1561,7 @@ void PhilipsPAR::ReadPAR(std::string parFile, struct par_parameter *pPar)
       // Only 1 slice, but how many repetitions of that slice?
       else
         {
+		//pPar->slicessorted = 1;
         int lineIncrement = 92;
         int echoIndex = 0;
         int cardiacIndex = 0;
@@ -1583,6 +1585,10 @@ void PhilipsPAR::ReadPAR(std::string parFile, struct par_parameter *pPar)
           }
         ++lineIncrement;
         tempInfo1 = GetImageInformationDefinitionV4(parFile, lineIncrement, this);
+		if ((tempInfo1.cardiac - tempInfo.cardiac) == 1 )
+          {
+			pPar->slicessorted = 1;
+		}
         while ( !tempInfo1.problemreading && tempInfo1.slice )
           {
           if ( slice == tempInfo1.slice )
diff --git a/Modules/IO/PhilipsREC/itk-module.cmake b/Modules/IO/PhilipsREC/itk-module.cmake
index 68b09ea..4cdb36b 100644
--- a/Modules/IO/PhilipsREC/itk-module.cmake
+++ b/Modules/IO/PhilipsREC/itk-module.cmake
@@ -11,7 +11,7 @@ itk_module(ITKIOPhilipsREC
     ITKTransform
     ITKImageGrid
     ITKImageIntensity
-  EXCLUDE_FROM_DEFAULT
+  #EXCLUDE_FROM_DEFAULT
   DESCRIPTION
     "${DOCUMENTATION}")
 # Extra dependency of ITKTransform is introduced by itkPhilipsRECImageIOOrientationTest.
