diff -urNb python-cmake-buildsystem-master/CMakeLists.txt Python-src/CMakeLists.txt
--- python-cmake-buildsystem-master/CMakeLists.txt	2014-08-21 23:21:20.000000000 +0200
+++ Python-src/CMakeLists.txt	2015-01-13 21:09:50.409871694 +0100
@@ -173,7 +173,7 @@ set(LIBPYTHON_VERSION ${PY_VERSION_MAJOR}.${PY_VERSION_MINOR})
 if(MSVC)
     set(LIBPYTHON_VERSION ${PY_VERSION_MAJOR}${PY_VERSION_MINOR})
 endif(MSVC)
-set(LIBPYTHON python${LIBPYTHON_VERSION})
+set(LIBPYTHON python${LIBPYTHON_VERSION}_mitk)
 
 # Proceed to the configure checks
 include(cmake/ConfigureChecks.cmake)
