diff --git a/CMakeOptions.cmake b/CMakeOptions.cmake
index 17fc5b8..23b3f1f 100644
--- a/CMakeOptions.cmake
+++ b/CMakeOptions.cmake
@@ -16,6 +16,19 @@ IF ( NOT VTK_FOUND )
 ENDIF ( NOT VTK_FOUND )
 
 #
+# Try to find GDCM and include its settings (otherwise complain)
+#
+IF ( NOT GDCM_FOUND )
+  FIND_PACKAGE(GDCM REQUIRED)
+  INCLUDE(${GDCM_USE_FILE})
+ENDIF ( NOT GDCM_FOUND )
+
+#
+# Try to find OpenCV which possibly is a dependency of ITK
+#
+FIND_PACKAGE(OpenCV)
+
+#
 # Try to find ITK and include its settings (otherwise complain)
 #
 IF ( NOT ITK_FOUND )

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 12c5a25..82d8f42 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,9 +17,11 @@ set(dirs
   Segmentation
   )
 
-IF (VTK_USE_RENDERING)
+OPTION(VTK_VMTK_RENDERING "Build and install classes in the vtkVmtk/Rendering directory." OFF)
+
+IF (VTK_VMTK_RENDERING)
   list(APPEND dirs Rendering)
-ENDIF (VTK_USE_RENDERING)
+ENDIF (VTK_VMTK_RENDERING)
 
 OPTION(VTK_VMTK_CONTRIB "Build and install classes in the vtkVmtk/Contrib directory." OFF)
 
@@ -69,9 +71,9 @@ INCLUDE_DIRECTORIES (${VTK_VMTK_SOURCE_DIR}/ComputationalGeometry)
 INCLUDE_DIRECTORIES (${VTK_VMTK_SOURCE_DIR}/DifferentialGeometry)
 INCLUDE_DIRECTORIES (${VTK_VMTK_SOURCE_DIR}/IO)
 INCLUDE_DIRECTORIES (${VTK_VMTK_SOURCE_DIR}/Misc)
-IF (VTK_USE_RENDERING)
+IF (VTK_VMTK_RENDERING)
   INCLUDE_DIRECTORIES (${VTK_VMTK_SOURCE_DIR}/Rendering)
-ENDIF (VTK_USE_RENDERING)
+ENDIF (VTK_VMTK_RENDERING)
 INCLUDE_DIRECTORIES (${VTK_VMTK_SOURCE_DIR}/Segmentation)
 IF (VTK_VMTK_CONTRIB)
   INCLUDE_DIRECTORIES (${VTK_VMTK_SOURCE_DIR}/Contrib)
   
diff --git a/Rendering/CMakeLists.txt b/Rendering/CMakeLists.txt
index 3665581..692384d 100644
--- a/RenderingCMakeLists.txt
+++ b/Rendering/CMakeLists.txt
@@ -5,6 +5,8 @@ if( ${VTK_VERSION_MAJOR} GREATER 5 )
     vtkCommonCore
     vtkInteractionWidgets
     vtkRenderingOpenGL
+	vtkRenderingFreeTypeOpenGL
+	vtkRenderingVolumeOpenGL
     )
   if (VTK_WRAP_PYTHON AND VTK_VMTK_WRAP_PYTHON)
     list(APPEND VTK_VMTK_RENDERING_COMPONENTS
